pkg/controller/rmdworkload/rmdworkload_controller_test.go

```diff
@@ -426,13 +426,38 @@ func TestRmdWorkloadControllerReconcile(t *testing.T) {
 			}
 		}))
 
+		// create a listener with the desired port.
+		address2 := "127.0.0.2:8080"
+		m, err := net.Listen("tcp", address2)
+		if err != nil {
+			t.Fatalf("Failed to create listener")
+		}
+
+		mux2 := http.NewServeMux()
+		mux2.HandleFunc("/v1/workloads/", (func(w http.ResponseWriter, r *http.Request) {
+			if r.Method == "GET" {
+				b, err := json.Marshal(tc.getWorkloadsResponse)
+				if err == nil {
+					fmt.Fprintln(w, string(b[:]))
+				}
+			} else {
+				if err == nil {
+					fmt.Fprintln(w, "ok")
+				}
+			}
+		}))
```

Comment:
Can this be automated like it is in `TestFindObsoleteWorkloads` and `TestFindTargetedWorkloads`? 
Or potentially extracting this functionality entirely into its own function and calling it from the individual tests. That would be a nice way to reduce duplicated code.
Maybe work for a future PR.

Reply:
I'll throw a story into my backlog to create a new function to automate listeners and probably work on that in the next sprint. 😊